go/vt/sqlparser/sql.y

```diff
@@ -602,19 +602,67 @@ base_select:
   }
 
 base_select_no_cte:
-  SELECT comment_opt cache_opt distinct_opt sql_calc_found_rows_opt straight_join_opt select_expression_list where_expression_opt group_by_opt having_opt window_opt
-  {
-    $$ = &Select{Comments: Comments($2), Cache: $3, Distinct: $4, Hints: $6, SelectExprs: $7, From: TableExprs{&AliasedTableExpr{Expr:TableName{Name: NewTableIdent("dual")}}}, Where: NewWhere(WhereStr, $8), GroupBy: GroupBy($9), Having: NewWhere(HavingStr, $10), Window: $11}
+  SELECT comment_opt cache_opt distinct_opt sql_calc_found_rows_opt straight_join_opt select_expression_list after_select
+  {
+    $8.(*Select).Comments = Comments($2)
+    $8.(*Select).Cache = $3
+    $8.(*Select).Distinct = $4
+    $8.(*Select).Hints = $6
+    $8.(*Select).SelectExprs = $7
+    $$ = $8
     if $5 == 1 {
       $$.(*Select).CalcFoundRows = true
     }
   }
-| SELECT comment_opt cache_opt distinct_opt sql_calc_found_rows_opt straight_join_opt select_expression_list FROM table_references where_expression_opt group_by_opt having_opt window_opt
+
+after_select:
```

Comment:
The reason I made separate rules for these is that `INTO` clause can be used in only one position but also can be absent. Then all of `FROM`, `WHERE`, `GROUP BY`, `HAVING` and `WINDOW` options are optional with empty case, so it creates shift/reduce on INTO being defined before and after these optional rules. To avoid this conflict, there needs to definite token after the first INTO to distinguish it from the second INTO option.

Reply:
Ahh... very interesting! I think it's worth spending some more time here and seeing if we can simplify somehow so that this will be easier to maintain/modify in the future. I checked out your branch and played with a bit and came up with an idea... it still needs to have this specific case tested, but I think it's worth trying out...

We could split `into_opt` into two non-terminals: one for the position after select expression list and one for the very end. I called that one at the end `into_end_opt`, but maybe there's a better name for it. From there, I was able to use `%prec` to explicitly tell yacc to lower the precedence of the `into_end_opt` match, after I defined `%right INTO` ahead of where `NULL` was already defined (to give it higher precedence). Then, I could do something like this: 

```yacc
into_opt:
%prec INTO
  {
    $$ = nil
  }
| INTO variable_list
  {
    $$ = $2
  }

into_end_opt:
%prec NULL
  {
    $$ = nil
  }
| INTO variable_list
  {
    $$ = $2
  }
```

Still needs to be testing more, but it's worth a shot to see if we can make that work and keep the grammar a little simpler. Precedence operators are usually used as a last resort when you can't refactor the rules cleanly, and this seems like a valid place to try them. 

btw... I've been experimenting with another MySQL grammar and just for fun, I looked up how they handled `select into` and they left a pretty interesting comment explaining their approach to handling this same shift/reduce conflict:
https://github.com/mysql/mysql-workbench/blob/8.0/library/parsers/grammars/MySQLParser.g4#L993

Essentially, they also didn't want to make the grammar rules more complicated and copy/paste around the select statement rules, so they chose to use precedence rules, as well.